### PR TITLE
Awards page

### PR DESCRIPTION
**Pull request by nastiaymlnv**
In AuthForm.jsx on row 91 for testing added navigating to the Awards page (and commenting navigation to home in row 92) because of missing the main page. It`s temporary action and it will be removed after adding the home page.

### CODE REVIEW

File: src/components/BaseModal/BaseModal.jsx [hunk 1]
```diff
@@ -34,15 +34,16 @@ export const BaseModal = ({setModalVisibility, children}) => {
...
-                    </CloseBtnImg>
-                </ModalContainerCloseBtnWrapper>
-                {children}
-            </ModalContainer>
-        </ModalBackground>
+        modalVisibility &&
+            <ModalBackground top={topOffset}>
+                <ModalContainer>
+                    <ModalContainerCloseBtnWrapper onClick={() => closeModalHandler()}>
+                        <CloseBtnImg src={closeCross}>
```
Comment (vladhybalo) [inline src/components/BaseModal/BaseModal.jsx]: CloseBtnImg -> CloseImg or CloseBtn)

File: src/components/BaseModal/BaseModal.jsx [hunk 2]
```diff
@@ -34,15 +34,16 @@ export const BaseModal = ({setModalVisibility, children}) => {
...
-        <ModalBackground top={topOffset}>
-            <ModalContainer>
-                <ModalContainerCloseBtnWrapper onClick={() => closeModalHandler()}>
-                    <CloseBtnImg src={closeCross}>
-                    </CloseBtnImg>
-                </ModalContainerCloseBtnWrapper>
-                {children}
-            </ModalContainer>
-        </ModalBackground>
+        modalVisibility &&
```
Comment (vladhybalo) [inline src/components/BaseModal/BaseModal.jsx]: For what case do you need this flag? As I see its possible to use modal without it. 
In parent you could use 
```
const [modalVisibility, setModalVisibility] = useState();
...
modalVisibility && <BaseModal setModalVisibility={setModalVisibility}/>
```

File: src/ducks/gifts/giftsActions.jsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    return {
+        type: GET_GIFTS,
+        payload: data.ruGifts
+    }
+}
+
+export const buyGifts = (data) => {
+    return {
+        type: BUY_GIFTS,
+        payload: data
```
Comment (vladhybalo) [inline src/ducks/gifts/giftsActions.jsx]: why not to call data - payload for shorter syntax

File: src/ducks/gifts/giftsActions.jsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    return dispatch => {
+        axios
+            .get(`https://kidslike-v1-backend.goit.global/gift`, {
+                headers: {
+                    'Authorization': token,
+                    'Content-Type': 'application/json'
+                }
+            })
+            .then(res => {
+                dispatch(getGifts(res.data));
```
Comment (vladhybalo) [inline src/ducks/gifts/giftsActions.jsx]: You could pass res.data.ruGifts here to simplify getGifts action creator

File: src/ducks/gifts/giftsActions.jsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            })
+            .catch(error => {
+                alert(error.message);
```
Comment (vladhybalo) [inline src/ducks/gifts/giftsActions.jsx:42]: shouldnt you use giftsErrorRequest here?

File: src/ducks/gifts/giftsReducer.jsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+const giftsReducer = (state = {}, action) => {
+    switch (action.type) {
+        case GET_GIFTS:
+            return {
+                gifts: action.payload
+            };
+        case BUY_GIFTS:
+            return {...state, purchasedGifts: action.payload};
+        case GIFTS_ERROR_REQUEST:
```
Comment (vladhybalo) [inline src/ducks/gifts/giftsReducer.jsx]: I think gifts_request_error sound more sophisticated and definitely.

File: src/middlewares/localStorageMiddleware.jsx [hunk 7]
```diff
@@ -1,5 +1,12 @@
+import { USER_SIGN_IN, USER_SIGN_UP } from "../ducks/userInfo/userInfoActions";
+
 export const localStorageMiddleware = (store) => (next) => (action) =>  {
+
     next(action);
+
+    if (action.type === USER_SIGN_IN || action.type === USER_SIGN_UP) {
+        localStorage.setItem("token", store.getState().userInfo.token);
+    }
```
Comment (vladhybalo) [inline src/middlewares/localStorageMiddleware.jsx]: You are right we need to save token for F5 case, but why do you choose localStorage?

File: src/pages/Awards/Awards.jsx [hunk 8]
```diff
@@ -1,10 +1,85 @@
...
-    return <Spinner></Spinner>
+    const dispatch = useDispatch();
+    let userAccess = useSelector(state => state.userInfo.token);
+    const gifts = useSelector(state => state.gifts.gifts);
+
+    if (!userAccess) {
+        userAccess = localStorage.getItem('token');
+    }
+
+    const [giftsList, setGiftsList] = useState([]);
```
Comment (vladhybalo) [inline src/pages/Awards/Awards.jsx]: I dont understand why do you need this variable, as you can directly use gifts from store?

File: src/pages/Awards/Awards.jsx [hunk 9]
```diff
@@ -1,10 +1,85 @@
...
+    ImproviseCard,
+    Checkbox,
+    SubmitButton
+} from "./Awards.style";
+
 const Awards = () => {
-    return <Spinner></Spinner>
+    const dispatch = useDispatch();
+    let userAccess = useSelector(state => state.userInfo.token);
+    const gifts = useSelector(state => state.gifts.gifts);
```
Comment (vladhybalo) [inline src/pages/Awards/Awards.jsx]: `gifts.gifts` sounds strange, doesn't it?

File: src/pages/Awards/Awards.jsx [hunk 10]
```diff
@@ -1,10 +1,85 @@
...
+    }
+
+    const [giftsList, setGiftsList] = useState([]);
+    const [giftsToBuyList, setGiftsToBuyList] = useState([]);
+    const [modalVisibility, setModalVisibility] = useState(false);
+
+    const addToBuyList = (id) => {
+        if (giftsToBuyList.includes(id)) {
+            let filteredArr = giftsToBuyList.filter(item => item !== id);
+            setGiftsToBuyList(filteredArr);
```
Comment (vladhybalo) [inline src/pages/Awards/Awards.jsx]: If its **add**ToBuyList why do you remove ids?)

File: src/pages/Awards/Awards.jsx [hunk 11]
```diff
@@ -1,10 +1,85 @@
...
+    }, [gifts]);
+
+    return (
+        <AwardsContainer>
+            <AwardsTitle />
+            <AwardsCardsContainer>
+                {
+                    giftsList.map(award =>
+                        {
+                            return (
```
Comment (vladhybalo) [inline src/pages/Awards/Awards.jsx]: if arrow function contains one statement `return` and `{}` brackets are redundant)

Comment (nastiaymlnv) [reply src/pages/Awards/Awards.jsx]: If remove brackets it throws an error

**Review by DmytroP888 — DISMISSED** [review]

File: src/middlewares/localStorageMiddleware.jsx [hunk 12]
```diff
@@ -1,5 +1,12 @@
+import { USER_SIGN_IN, USER_SIGN_UP } from "../ducks/userInfo/userInfoActions";
+
 export const localStorageMiddleware = (store) => (next) => (action) =>  {
+
     next(action);
+
+    if (action.type === USER_SIGN_IN || action.type === USER_SIGN_UP) {
+        sessionStorage.setItem("token", store.getState().userInfo.token);
```
Comment (vladhybalo) [inline src/middlewares/localStorageMiddleware.jsx]: 🤔

File: src/pages/Awards/Awards.jsx [hunk 13]
```diff
@@ -1,10 +1,75 @@
...
+    AwardsCardsContainer,
+    ImproviseCard,
+    Checkbox,
+    SubmitButton
+} from "./Awards.style";
+
 const Awards = () => {
-    return <Spinner></Spinner>
+    const dispatch = useDispatch();
+    let userAccess = useSelector(state => state.userInfo.token);
```
Comment (vladhybalo) [inline src/pages/Awards/Awards.jsx]: pls forget about `let`

File: src/pages/Awards/Awards.jsx [hunk 14]
```diff
@@ -1,10 +1,75 @@
...
+    const gifts = useSelector(state => state.awards.gifts);
+
+    const [giftsToBuyList, setGiftsToBuyList] = useState([]);
+    const [modalVisibility, setModalVisibility] = useState(false);
+
+    const modifyBuyList = (id) => {
+        if (giftsToBuyList.includes(id)) {
+            let filteredArr = giftsToBuyList.filter(item => item !== id);
```
Comment (vladhybalo) [inline src/pages/Awards/Awards.jsx]: https://github.com/vladhybalo/kids-awards/pull/16/files#r952536691

File: src/pages/Awards/Awards.jsx (hunk 11 above)
Comment (vladhybalo) [reply src/pages/Awards/Awards.jsx]: Impossible)

File: src/pages/Awards/Awards.style.jsx [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+`;
+
+export const AwardsCardsContainer = styled.ul`
+    margin-top: 40px;
+    margin-bottom: 60px;
+`;
+
+export const ImproviseCard = styled.li`
+    /* margin-top: 40px;
+    margin-bottom: 60px; */
```
Comment (vladhybalo) [inline src/pages/Awards/Awards.style.jsx]: to remove

File: src/store.jsx [hunk 16]
```diff
@@ -6,9 +6,21 @@ import reducers from "./ducks";
...
+    }
+    else {
+        return {};
+    }
+}
+
 const getInitialStore = () => {
     return {
-        userInfo: {}
+        userInfo: getInitialStateToken(),
```
Comment (vladhybalo) [inline src/store.jsx]: Actualy, you get here all `userInfo` section, not only token

File: src/ducks/gifts/giftsActions.jsx [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+        payload: data
+    }
+}
+
+export const getGiftsData = (token) => {
+    return dispatch => {
+        axios
+            .get(`https://kidslike-v1-backend.goit.global/gift`, {
+                headers: {
+                    'Authorization': token,
```
Comment (VladyslavHybalo) [inline src/ducks/gifts/giftsActions.jsx]: You can get store from state here, and no need to pass it from component

File: src/store.jsx [hunk 18]
```diff
@@ -6,9 +6,37 @@ import reducers from "./ducks";
 
 import localStorageMiddleware from "./middlewares/localStorageMiddleware";
 
+const getCookie = (cookieName) => {
```
Comment (VladyslavHybalo) [inline src/store.jsx]: I think this function is very usefull and could be used anywhere in the project, so it makes sense to move it to utils

**Review by VladyslavHybalo — APPROVED** [review]

**Review by andriiivaniuk — APPROVED** [review]

**Review by vladhybalo — APPROVED** [review]

**Pull request merged**